arc_cache.go

```diff
@@ -145,20 +202,35 @@ func (b *arccache) Put(bl blocks.Block) error {
 }
 
 func (b *arccache) PutMany(bs []blocks.Block) error {
+	mxs := [256]*sync.Mutex{}
 	var good []blocks.Block
 	for _, block := range bs {
 		// call put on block if result is inconclusive or we are sure that
 		// the block isn't in storage
 		if has, _, ok := b.queryCache(block.Cid()); !ok || (ok && !has) {
+			if !block.Cid().Defined() {
+				log.Error("undefined cid in arc cache")
+				continue
```

Comment:
unsure if we want to continue or bail in the event an undefined CID is in `bs`, thoughts?

Reply:
On put, bail. This is definitely an invalid block.

Reply:
removed this case since it should never (right?) be the case that the CID of a blocks.Block is undefined.

Reply:
Yeah, that should be impossible.